Jenkinsfiles/IT-release-on-demand-packages.groovy

```diff
@@ -8,17 +8,53 @@ class ReleaseBuild implements Serializable {
+    timeout(300) {
+        def releaseJob = "Deploy/" + (params.RELEASE_TYPE == "release" ? "IT-release-on-demand-build" : "IT-nuxeo-master-build")
 
-                checkout([$class: 'GitSCM', branches: [[name: BRANCH]], browser: [$class: 'GithubWeb', repoUrl: 'https://github.com/nuxeo/nuxeo'], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'nuxeo'], [$class: 'PerBuildTag'], [$class: 'WipeWorkspace'], [$class: 'CloneOption', depth: 0, noTags: false, reference: '', shallow: false, timeout: 60], [$class: 'CheckoutOption', timeout: 60]], submoduleCfg: [], userRemoteConfigs: [[url: 'git@github.com:nuxeo/nuxeo.git']]])
-                step([$class: 'CopyArtifact', filter: 'release-nuxeo.log', fingerprintArtifacts: true, projectName: releaseBuild.projectName, selector: [$class: 'SpecificBuildSelector', buildNumber: releaseBuild.buildNumber]])
+        node('SLAVE') {
+            stage('clone') {
+                def releaseBuild = new ReleaseBuild(input(message: 'Select the distribution to release against:', parameters: [
+                        [$class: 'RunParameterDefinition', filter: 'SUCCESSFUL', name: 'RELEASE_BUILD', projectName: releaseJob]
+                ]))
+                checkout([
+                        $class: 'GitSCM',
+                        branches: [[name: checkForBranch('nuxeo', 'nuxeo', params.BRANCH, 'master')]],
+                        browser: [$class: 'GithubWeb', repoUrl: 'https://github.com/nuxeo/nuxeo'],
+                        extensions: [
+                                [$class: 'RelativeTargetDirectory', relativeTargetDir: 'nuxeo'],
+                                [$class: 'PerBuildTag'],
+                                [$class: 'WipeWorkspace'],
+                                [$class: 'CloneOption', depth: 0, noTags: false, reference: '', shallow: false, timeout: 60],
+                                [$class: 'CheckoutOption', timeout: 60]
+                        ],
+                        userRemoteConfigs: [[url: 'git@github.com:nuxeo/nuxeo.git']]
+                ])
+                step([
+                        $class: 'CopyArtifact',
+                        filter: 'release-nuxeo.log',
+                        fingerprintArtifacts: true,
+                        projectName: releaseBuild.projectName,
+                        selector: [$class: 'SpecificBuildSelector', buildNumber: releaseBuild.buildNumber]
+                ])
                 withCredentials([usernamePassword(credentialsId: 'eea4e470-2c5e-468f-ab3a-e6c81fde94c0', passwordVariable: 'GITHUB_PASSWD', usernameVariable: 'GITHUB_TOKEN')]) {
-                    sh 'curl -u "$GITHUB_TOKEN:$GITHUB_PASSWD" -H "Accept: application/vnd.github.v3.raw" -o packages.ini $MARKETPLACE_INI?ref=$BRANCH'
+                    sh('curl -u "$GITHUB_TOKEN:$GITHUB_PASSWD" -H "Accept: application/vnd.github.v3.raw" -o packages.ini $MARKETPLACE_INI?ref=' +
```

Comment:
why not `curl -n` instead of `-u`? passing credentials as clear env params is not good

Reply:
Those env params only exist during the sh/curl step and are not exposed to the rest of the build + I though it was a better practice to rely on Jenkins credentials rather than a config file on the slave.
Note that the credentials are safely replaced by *** on the build console.